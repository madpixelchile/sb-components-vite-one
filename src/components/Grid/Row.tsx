import { AnyProp, ChildrenProp, ClassNameProp } from "../../types/types";
import { StyledGrid } from "./StyledGrid";

export interface Props extends 
ChildrenProp, 
ClassNameProp, 
AnyProp {}

export const Row = ( { children = null, className = '', ...props } : Props) => {
  return (
    <>
        {  <StyledGrid className={`row ${className}`} {...props} >{ children }</StyledGrid>  }
    </>
  )
}

export default Row;

