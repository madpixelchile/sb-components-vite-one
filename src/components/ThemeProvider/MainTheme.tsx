

import { ThemeProvider } from 'styled-components';
import { StyledThemeProvider } from './StyledThemeProvider';
import { darkMode, lightMode } from '../../styles/theme';

export interface MainThemeProps {
  children?: React.ReactNode;
  theme?: string | {},
  [key: string]: any;
}

export const MainTheme = ({ children = null, theme = 'light' }:MainThemeProps) => {

  let selectedTheme: string | object = lightMode;

  if(typeof theme !== 'object'){
    if(theme === 'light'){
      //No object - string light - default lightModeAdded
      selectedTheme = lightMode;
    }else if(theme === 'dark'){
      //No object - string dark -  default darkModeAdded
      selectedTheme = darkMode;
    }
  }else if(typeof theme === 'object'){
    if(Object.keys(theme).length === 0){ //!PENDING Compare with the final BASE required properties
      //Empty object - Default lightMode added
      selectedTheme = lightMode;
    }else{
      //Full object - CustomMode or theme added
      selectedTheme = theme;
    }
  }

  return (
    <>
        <ThemeProvider theme={ selectedTheme }>
            <StyledThemeProvider/>
            { children }
        </ThemeProvider>
    </>
  )
}

export default  MainTheme;
