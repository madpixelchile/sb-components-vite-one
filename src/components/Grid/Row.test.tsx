
import { render } from '@testing-library/react'; //screen
import { Row } from './Row';
import MainTheme from '../ThemeProvider/MainTheme';

describe('Testing Rows', () => {

    it('Testing props', () => {

        const { getByText, container } = render(<MainTheme><Row className='no-gutters'>Children</Row></MainTheme>);
        const childrenElement = getByText('Children');
        expect(childrenElement.innerHTML).toBe('Children');

        const classProp = container.getElementsByClassName('no-gutters');
        expect(classProp.length).toBe(1)

        // screen.debug();

        expect(container).toMatchSnapshot();

    })

})