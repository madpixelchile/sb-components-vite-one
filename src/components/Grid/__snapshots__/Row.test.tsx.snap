// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Rows Testing props 1`] = `
<div>
  <div
    class="sc-bdnyFh ktBSwJ row no-gutters"
  >
    Children
  </div>
</div>
`;
